packages/ListBox/src/components/Option/helpers/optionState.js

```diff
@@ -33,7 +33,8 @@ export function getDataOptions(children) {
   let index = 0;
 
   React.Children.toArray(children).forEach(child => {
-    if (child.type && isWhiteListed(child.type.displayName)) {
+    const type = child.type.type || child.type;
```

Comment:
This covers both memoized components (Option component) or non memoized components, eg divider etc